crates/cast/src/starknet_commands/account/deploy.rs

```diff
@@ -30,9 +35,143 @@ pub struct Deploy {
+            chain_id,
+            max_fee,
+            wait,
+            class_hash,
+        )
+        .await
+    }
+}
+
+async fn deploy_from_keystore(
+    provider: &JsonRpcClient<HttpTransport>,
+    chain_id: FieldElement,
+    max_fee: FieldElement,
+    wait: bool,
+    keystore_path: Utf8PathBuf,
+    account_path: Utf8PathBuf,
+) -> Result<InvokeResponse> {
+    let contents =
+        std::fs::read_to_string(account_path.clone()).context("Couldn't read account file")?;
+    let mut items: Map<String, serde_json::Value> = serde_json::from_str(&contents)
+        .map_err(|_| anyhow!("Failed to parse account file at {account_path}"))?;
+
+    if items["deployment"].is_null() {
+        bail!("No deployment field in account JSON file");
+    }
+
+    let status = items
+        .get("deployment")
+        .and_then(|deployment| deployment.get("status"))
+        .and_then(serde_json::Value::as_str)
+        .ok_or_else(|| anyhow::anyhow!("Failed to get status from account JSON file"))?;
+
+    if status == "deployed" {
+        bail!("Account already deployed");
+    }
+
+    let salt = FieldElement::from_hex_be(
+        items
+            .get("deployment")
+            .and_then(|deployment| deployment.get("salt"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("Failed to get salt from account JSON file"))?,
+    )?;
+    let oz_class_hash = FieldElement::from_hex_be(
+        items
+            .get("deployment")
+            .and_then(|deployment| deployment.get("class_hash"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("Failed to get salt from account JSON file"))?,
+    )?;
+
+    let private_key = SigningKey::from_keystore(keystore_path, get_keystore_password()?.as_str())?;
+    let public_key: FieldElement = {
+        let pk = items
+            .get("variant")
+            .and_then(|v| v.get("public_key"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("No public_key in account JSON file"))?;
+        parse_number(pk)?
+    };
+    if public_key != private_key.verifying_key().scalar() {
+        bail!("Public key and private key from keystore do not match");
+    }
+
+    items["deployment"]["status"] = serde_json::Value::from("deployed");
+    items.get_mut("deployment").and_then(|deployment| {
+        deployment
+            .as_object_mut()
+            .expect("should be an object")
+            .remove("salt")
+    });
+    items["deployment"]["address"] = format!(
+        "{:#x}",
+        get_contract_address(
+            salt,
+            oz_class_hash,
+            &[private_key.verifying_key().scalar()],
+            FieldElement::ZERO,
+        )
+    )
+    .into();
+
+    let result = deploy_oz_account(
+        provider,
+        oz_class_hash,
+        private_key,
+        salt,
+        chain_id,
+        max_fee,
+        wait,
+    )
+    .await?;
+
```

Comment:
it would be cool if this could be more atomic. deploy_oz_account could fail, leaving us with an account file that says the account was deployed, even though it wasn't. There are multiple ways we could approach this problem, let me know if you need any help figuring this out.

Reply:
🤔 not sure I am following - if `deploy_oz_account` fails the error gets propagated through `?` and no changes in files are made

Reply:
yes you're right, sorry, brainfart
<img width="387" alt="obraz" src="https://github.com/foundry-rs/starknet-foundry/assets/9470712/88dd8f03-6a67-4fda-bbaa-c6aa0542fed7">